### ISSUE TITLE
Careful considerations of logical operations following conditional block

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15261]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/IfUnlessModifier

### Steps to reproduce the problem

```ruby
#!/usr/bin/env ruby

$need = false
$load = nil
$file = 'file'

def original
  # IfUnlessModifier
  if $need
    $load
  end || $file
end

def one_nop_wrong
  $load || $file if $need # Pitfall undetectable
end

def one_par_right
  ($load if $need) || $file
end

def two_nop_ambig
  $need && $load || $file # Pitfall caught by AmbiguousOperatorPrecedence
end

def two_par_right
  ($need && $load) || $file
end

pp original
pp one_nop_wrong
pp one_par_right
pp two_nop_ambig
pp two_par_right
```

### Expected behavior

1. Update text telling coder to very careful rearranging of operators and placement of parentheses.
2. Really don't flash cop at all if block has code tailing `end`, because then the rewrite is not so obvious anymore.

### Actual behavior

```
1. Favor modifier `if` usage when having a single-line body.
2. Another good alternative is the usage of control flow `&&`/`||`
```

Autocorrect _does_ produce parenthesized code, but manual instruction fails to mention parentheses.

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-06-09 bd81430c6
```

**Comment by rafa-el-souza** [comment]
Hi y'all! I'm interested in contributing. I reproduced the issue with the current master code, and the thing is `rubocop -a` **does** wrap the expression in parens when the `if`'s parent is an operator:

https://github.com/rubocop/rubocop/blob/5e56132d85d54705bf0618dce3599d04032142b1/lib/rubocop/cop/mixin/statement_modifier.rb#L91-L101

So the issue only occurs when fixing manually, and because the message does not mention that parens are necessary. I believe the solution here would be having a different message for when `parenthesize? == true`. Something like: `Favor modifier if usage when having a single-line body. Wrap the expression in parentheses to keep the current behavior, as it is part of a larger expression.`

About `Really don't flash cop at all if block has code tailing end, because then the rewrite is not so obvious anymore.`: since `rubocop -a` works as intended and there is deliberate parenthesization logic in place confirmed by specs (e.g. "adds parentheses so as not to change meaning"), that seems like the wrong way to go:

https://github.com/rubocop/rubocop/blob/5e56132d85d54705bf0618dce3599d04032142b1/spec/rubocop/cop/style/if_unless_modifier_spec.rb#L688-L716

**Comment by forthrin** [comment]
Good observations and suggestions. Yes, autocorrect does the right thing. A possible helper text could be:

`This statement can be rewritten as a single line using carefully placed parentheses and logical operators.`

**Comment by bbatsov** [comment]
Your analysis is right on. The autocorrect already parenthesizes correctly, so this is purely a messaging gap - a dedicated message when `parenthesize?` is true (calling out that the parentheses are needed to preserve behavior) is exactly the fix. And agreed on not suppressing the cop just because there's code after the `end`; the autocorrect handles that case deliberately. Would happily take a PR - go for it, and ping me if you get stuck.
